test/javaTests/src/org/criu/java/tests/CheckpointRestore.java

```diff
@@ -0,0 +1,345 @@
+package org.criu.java.tests;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.io.*;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileChannel.MapMode;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+
+public class CheckpointRestore {
+	private MappedByteBuffer mappedByteBuffer = null;
+	private String testName = "";
+	private String logFolder = Helper.LOG_FOLDER + "/";
+	private String outputFolder = Helper.OUTPUT_FOLDER_NAME + "/";
+
+	/**
+	 * Create CRlog and output directory if they don't exist.
+	 * Delete failed dump and restore logs if they exist.
+	 *
+	 * @throws IOException
+	 */
+	@BeforeSuite
+	void suiteSetup() throws IOException {
+	System.out.println("Tests are to be run as root user!!");
```

Comment:
This is slightly misleading. If you run the tests as *root* user inside an unprivileged container are they going to fail?

Reply:
I have cleared this up, and mentioned the capabilities required by the user running these tests in README.